### DISCUSSION TITLE
Batch updates for attomWithImmer

### DISCUSSION THREAD

**Discussion by alexnix** [pmndrs/jotai#2223] [Bug report]
I have an `atomWithImmer` and on a button press I call the setter method twice. Instead of getting a single render, as expected with `useState` I get two renders. I could look into merging the two updates and call the setter only once, but is there a reason for witch the two are not batched anyway ?

**Comment by dai-shi** [comment]
Yes, we basically do all updates in sync. So, not batching is intentional. React will batch updates anyway. So, keeping the behavior of `useState` is the major goal.
You can create a special atom on your own to batch updates. I think someone did atomWithDebounce, and the approach could be similar.

**Comment by alexnix** [comment]
> React will batch updates anyway

So then should't I get only one render even tho I am calling the atom setter two times?

**Reply by dai-shi** [reply]
I'm not very confident about `atomWithImmer`. Can you try with the normal `atom` and see if there's a difference?
